src/qibolab/serialize.py

```diff
@@ -131,11 +136,13 @@ def dump_qubits(
     characterization = {
         "single_qubit": {q: qubit.characterization for q, qubit in qubits.items()},
     }
+
+    kernels = Kernels()
     for q in qubits:
         qubit = characterization["single_qubit"][q]
-        kernel_path = qubit["kernel_path"]
-        if kernel_path is not None:
-            qubit["kernel_path"] = kernel_path.name
+        if qubit["kernel"] is not None:
+            kernel = qubit.pop("kernel")
+            kernels[q] = kernel
```

Comment:
I think this is a bit obscure. We are putting the kernel array from the `Qubit` object to the characterization dict and then popping it out. I think it makes more sense to put `"kernel"` in the `EXCLUDED_FIELDS` above the `Qubit` object so that it is not included in the `characterization` dict and here just grab it from the `Qubit` directly:
```suggestion
        qubit = qubits[q]
        if qubit.kernel is not None:
            kernels[q] = qubit.kernel
```